src/tree.js

```diff
@@ -27,14 +27,23 @@ const { assign, defineProperties } = Object;
  * @param {Array<any>} args
  */
 const defaultMiddleware = (localMicrostate, transition, args) => {
-  let tree = reveal(localMicrostate);
+  let { lens, root } = reveal(localMicrostate);
 
-  let { microstate } = tree.apply(focus => {
+  let applyTransition = focus => {
     let next = transition.apply(focus.microstate, args);
-    return next instanceof Microstate ? reveal(next) : focus.assign({ data: { value: next }});
-  });
+    if (next instanceof Microstate) {
+      return reveal(next);
+    } else {
+      return focus.assign({ data: { value: next }});
+    }
+  };
+
+  // transition is applied to the end of the path
+  // it returns the next tree which is returned
+  let nextTree = over(lens, applyTransition, root);
 
-  return microstate;
+  // map tree to make sure that it gets the root applied
+  return map(tree => tree, nextTree).microstate;
```

Comment:
Tangential: I feel as though the requirement we have to maintain the root context within a tree is an albatross.